SimPEG/EM/Static/DC/ProblemDC.py

```diff
@@ -37,14 +39,55 @@ def fields(self, m=None):
         f[Srcs, self._solutionType] = u
         return f
 
-    def Jvec(self, m, v, f=None):
+    def getJ(self, m, f=None):
+
+        if self.verbose:
+            print("Calculating J and storing")
+
+        self.model = m
 
         if f is None:
             f = self.fields(m)
 
+        self.Jmat = []
+        AT = self.getA()
+
+        for src in self.survey.srcList:
+            u_src = f[src, self._solutionType]
+            for rx in src.rxList:
+                # wrt f, need possibility wrt m
+                PT = rx.getP(self.mesh, rx.projGLoc(f)).toarray().T
+                df_duTFun = getattr(f, '_{0!s}Deriv'.format(rx.projField),
+                                    None)
+                df_duT, df_dmT = df_duTFun(src, None, PT, adjoint=True)
+
+                ATinvdf_duT = self.Ainv * df_duT
+
+                dA_dmT = self.getADeriv(u_src, ATinvdf_duT, adjoint=True)
+                dRHS_dmT = self.getRHSDeriv(src, ATinvdf_duT, adjoint=True)
+                du_dmT = -dA_dmT + dRHS_dmT
+                Jt = (df_dmT + du_dmT).astype(float)
+
+                self.Jmat.append(np.vstack(Jt))
+
+        self.Jmat = np.hstack(self.Jmat).T
+        return self.Jmat
+
+    def Jvec(self, m, v, f=None):
+
+        if self.storeJ:
+            if self.Jmat is None:
+                if f is None:
+                    self.model = m
+                    f = self.fields(m)
+                self.getJ(m, f=f)
+            return Utils.mkvc(np.dot(self.Jmat, v))
+
         self.model = m
 
-        # Jv = self.dataPair(self.survey)  # same size as the data
+        if f is None:
+            f = self.fields(m)
+
         Jv = []
 
         A = self.getA()
```

Comment:
it doesn't look like `A` is used in the below calculation?

Reply:
Nope. Thanks for catching this.